packages/flutter_tools/test/general.shard/devfs_test.dart

```diff
@@ -197,6 +198,7 @@ void main() {
       );
 
       expect(report.success, false);
+      expect(devFS.lastCompiled, previousCompile);
```

Comment:
So this test checks that the correct value of `lastCompiled` is retained, but it would probably also be good to have a test that demonstrates why retaining the correct value is necessary.

Reply:
I've updated with a test case that confirms that the lastCompiled time is updated on success. Correctly using that value is a bit out of scope, since that is only used by the resident_runner